test/lifecycles/CustomComponentLifecycle.test.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2024 Verji Tech AS
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from "react";
+import { render, screen } from "@testing-library/react";
+
+import { RuntimeModule } from "../../src/RuntimeModule";
+import {
+    CustomComponentLifecycle,
+    CustomComponentListener,
+    CustomComponentOpts,
+} from "../../src/lifecycles/CustomComponentLifecycle";
+
+//Mock a CustomUserMenu
+class CustomUserMenu extends React.Component {
+    render() {
+        return (
+            <div className="customUserMenu" title="CustomUserMenu">
+                {this.props.children}
+            </div>
+        );
+    }
+}
+//Mock a default UserMenu
+class UserMenu extends React.Component {
+    render() {
+        return (
+            <div className="defaultUserMenu" title="UserMenu">
+                {this.props.children}
+            </div>
+        );
+    }
+}
+describe("CustomComponentLifecycle", () => {
+    let module: RuntimeModule;
+
+    beforeAll(() => {
+        module = new (class extends RuntimeModule {
+            constructor() {
+                super(undefined as any);
+
+                this.on(CustomComponentLifecycle.UserMenu, this.customComponentListener);
+            }
+
+            protected customComponentListener: CustomComponentListener = (customComponentOpts: CustomComponentOpts) => {
+                customComponentOpts.CustomComponent = ({ children }) => {
+                    let usermenu: any = React.Children.toArray(children)[0];
```

Comment:
All of this casting and use of `any` feels not great. Is there some way we can get more type-safety when customizing like this? (again this goes back to not knowing how much type-safety to expect in this project)

Reply:
Just to be clear, you are referring to: 
1. `super(undefined as any);`
and 
2. `let usermenu: any = React.Children.toArray(children)[0];` ? 

In case 1. seems to be the standard of all other lifecycle tests within this project. But I can look into it - should be possible to use the interface for ModuleApi instead. 

In case 2. Is just a mock-implementation of what a module might look like. But I'll look into it to see if I can get rid of the explicit any.  And see what I can do to provide more type-safety. 

Appreciate the feedback @robintown! 👍